src/server/deviceDescriptorsSource.json

```diff
@@ -1140,5 +1140,95 @@
     "isMobile": true,
     "hasTouch": true,
     "defaultBrowserType": "chromium"
+  },
+  "Surface Go 2 10.5-inch": {
+    "userAgent": "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/93.0.4543.0 Safari/537.36",
+    "screen": {
+      "width": 1920,
+      "height": 1280
+    },
+    "viewport": {
+      "width": 1280,
+      "height": 720
+    },
+    "deviceScaleFactor": 1,
+    "isMobile": false,
+    "hasTouch": true,
+    "defaultBrowserType": "chromium"
+  },
+  "Surface Pro X 13-inch": {
+    "userAgent": "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/93.0.4543.0 Safari/537.36 Edg/91.0.864.37",
+    "screen": {
+      "width": 1440,
+      "height": 960
+    },
+    "viewport": {
+      "width": 1280,
+      "height": 720
+    },
+    "deviceScaleFactor": 2,
+    "isMobile": false,
+    "hasTouch": true,
+    "defaultBrowserType": "chromium"
+  },
+  "Surface Book 3 15-inch": {
+    "userAgent": "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/93.0.4543.0 Safari/537.36 Edg/91.0.864.37",
+    "screen": {
+      "width": 1620,
+      "height": 1080
+    },
+    "viewport": {
+      "width": 1280,
+      "height": 720
+    },
+    "deviceScaleFactor": 2,
+    "isMobile": false,
+    "hasTouch": true,
+    "defaultBrowserType": "chromium"
+  },
+  "MacBook Air 2020 13-inch": {
+    "userAgent": "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/605.1.15 (KHTML, like Gecko) Version/14.1.1 Safari/605.1.15",
+    "screen": {
+      "width": 1440,
+      "height": 900
+    },
+    "viewport": {
+      "width": 1280,
+      "height": 720
+    },
+    "deviceScaleFactor": 2,
+    "isMobile": false,
+    "hasTouch": false,
+    "defaultBrowserType": "webkit"
+  },
+  "MacBook Pro 2020 13-inch": {
+    "userAgent": "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/605.1.15 (KHTML, like Gecko) Version/14.1.1 Safari/605.1.15",
+    "screen": {
+      "width": 1440,
+      "height": 900
+    },
+    "viewport": {
+      "width": 1280,
+      "height": 720
+    },
+    "deviceScaleFactor": 2,
+    "isMobile": false,
+    "hasTouch": false,
+    "defaultBrowserType": "webkit"
+  },
+  "MacBook Pro 2019 16-inch": {
```

Comment:
I think this does not solve the problem of running tests with "MacBook Chrome", "Surface Chrome", "MacBook Safari", "Surface Firefox", "Android Chrome" and "iPhone Safari". Any ideas?

Reply:
When using the test-runner with a device, we automatically use the correct one via the `defaultBrowserType` deviceDescriptor property and pass it over to the browserName. Also in theory we should not prevent users from testing the devices in other browsers, or do I miss something?